### PR TITLE
Make ParsedMediaType truly immutable

### LINKED ISSUE #67545: RestGetAliasesActionTests failures
**Build scan**: https://gradle-enterprise.elastic.co/s/5mihtfdkcxka4 (my machine), https://gradle-enterprise.elastic.co/s/mp56dimbgyk4o, https://gradle-enterprise.elastic.co/s/ljmyx42y466go

**Repro line**: ```./gradlew ':server:test' --tests "org.elasticsearch.rest.action.admin.indices.RestGetAliasesActionTests.testAliasWildcardsIncludeAndExcludeSome" -Dtests.seed=FA876308D81E4E55 -Dtests.security.manager=true -Dtests.locale=sk-SK -Dtests.timezone=Europe/Tallinn -Druntime.java=15```

**Reproduces locally?**: Not with the above, but it seems pretty deterministic (or at least quite frequent) for me with just `./gradlew ':server:test'`.

**Applicable branches**: `master`

**Failure history**: https://build-stats.elastic.co/goto/3a837e6a1f9f002ea6e70bca9e6c257d

**Failure excerpt**:

```
org.elasticsearch.rest.action.admin.indices.RestGetAliasesActionTests > testAliasWildcardsIncludeAndExcludeSome FAILED
    java.lang.AssertionError:
    Expected: "application/json"
         but: was "application/json;charset=utf-8"
        at __randomizedtesting.SeedInfo.seed([FA876308D81E4E55:731102B1343B21F5]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.rest.action.admin.indices.RestGetAliasesActionTests.testAliasWildcardsIncludeAndExcludeSome(RestGetAliasesActionTests.java:101)
```

Looks like it's probably related to #67409, /cc @pgomulka

### PR DESCRIPTION
**Pull request by pgomulka**
By accident ParsedMediaType was not truly immutable and it was possible
to modify other ParsedMediaType's by accident when building.
ParsedMediaType#parseMediaType(XContentType,Map) was modifying
ParsedMediaTypes within XContentType. This should never happen.

This commit is addressing this by making Map of parameters immutable.

closes #67545

### CODE REVIEW

File: libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java [hunk 1]
```diff
@@ -108,8 +108,9 @@ public static ParsedMediaType parseMediaType(String headerValue) {
 
     public static ParsedMediaType parseMediaType(XContentType requestContentType, Map<String, String> parameters) {
         ParsedMediaType parsedMediaType = requestContentType.toParsedMediaType();
-        parsedMediaType.parameters.putAll(parameters);
-        return parsedMediaType;
+
+        return new ParsedMediaType(parsedMediaType.originalHeaderValue,
+            parsedMediaType.type, parsedMediaType.subType, parameters);
```
Comment (joegallo) [inline libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java:113]: I dunno how paranoid we oughta be here, but I think this one factory method would still leak mutability. Somebody could pass in a parameters map, and then modify it later, IIUC the `Collections.unmodifiableMap` above only enforces an immutable view, if the underlying map is modified, it doesn't do anything for us.

`this.parameters = Map.copyOf(parameters);` above would protect against this, too, right?

Comment (joegallo) [reply libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java:112]: Fixed in 6614ef1

**Review by joegallo — APPROVED** [review]
LGTM, thanks!

File: libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java (hunk 1 above)
Comment (pgomulka) [reply libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java:112]: @joegallo this is great point. you are right -  `unmodifiableMap` is just an immutable view. copyOf is better there. Thanks for the fix!!!

**Pull request merged**